libs/wingsdk/src/target-tf-azure/bucket.inflight.ts

```diff
@@ -199,8 +201,18 @@ export class BucketClient implements IBucketClient {
    * @Throws if the file is not public or if object does not exist.
    */
   public async publicUrl(key: string): Promise<string> {
-    this._public; // a little help for implementing public_url later on
-    throw new Error(`publicUrl is not supported yet. (key=${key})`);
+    if (!this._public) {
+      throw new Error("Cannot provide public url for a non-public bucket");
+    }
+    if (!(await this.exists(key))) {
+      throw new Error(
+        `Cannot provide public url for a non-existent key (key=${key})`
+      );
+    }
+
+    return encodeURI(
+      `https://${this.storageAccount}.blob.core.windows.net/${this.bucketName}/${key}`
```

Comment:
@hm-harshit I think its fine to leave this as `bucketName` for now. 

Since in the sdk we chose to use the term `bucket` as the abstract term to refer to (aws-buckets, azure-container, gcp-bucket, etc..) so the client itself expects `bucketName` as an argument to the client.

I.E. in Winglang the below code is reference an abstract class that when compiled to a target will become a concrete version of (aws-bucket, azure-container, gcp-bucket, etc.)
```js
let b = new Bucket(); 
```

It may make sense for a refactor down the line but I believe that is out of scope for this PR.